### PR TITLE
Fallback chapter name if it ends up as blank

### LINKED ISSUE #8810: Chapter renamed by number, blank or (invalid)
### Steps to reproduce

1. Download comic
2. Chapter name is from number
3. Or chapter folder named (invalid) or blank

### Expected behavior

Chapter have "Chapter" attribute on first
Like "Chapter 01"

### Actual behavior

Chapter renamed by number and in the storage became (invalid) if not have a name or blank

### Crash logs

_No response_

### Tachiyomi version

0.14.2

### Android version

Android 11

### Device

Realme 7i

### Other details

Screenshot_2022-12-27-21-48-38-95_afd66b7916d19a60815f56273408b73e
Screenshot_2022-12-27-21-49-07-46_83ab3b20d3fba5c9c905d050db1f8de3

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.14.2](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by Hipp0x**
fix #8810 

Summary :
I changed the directory name when we download a manga which chapters name only composed of numbers or spaces.
It will appear like "Chapter 5" instead of "5" in the download directory.

### CODE REVIEW

**Comment by Hipp0x** [comment]
Can someone review my PR ?

File: app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadProvider.kt [hunk 1]
```diff
@@ -138,14 +138,35 @@ class DownloadProvider(
...
+     * Return the new name for the chapter (in case it's only a number or blanck)
+     *
+     * @param chapterName the name of the chapter
+     */
+    private fun getNewChapterName(chapterName: String): String {
+        var newChapterName = chapterName
+        try {
+            val chapterToInt = Integer.parseInt(chapterName)
+            if (chapterToInt.toString() == chapterName) {
+                newChapterName = "${R.string.chapter} $chapterName"
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadProvider.kt:160]: `R.string.chapter` resolves to some number, not the string. You'd need to do `context.getString(R.string.chapter)`.

That said, there's no point in localizing this, it'd break downloads if you switched the app language of the translation changes for some reason.

File: app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadProvider.kt [hunk 2]
```diff
@@ -138,14 +138,35 @@ class DownloadProvider(
...
 
+    /**
+     * Return the new name for the chapter (in case it's only a number or blanck)
+     *
+     * @param chapterName the name of the chapter
+     */
+    private fun getNewChapterName(chapterName: String): String {
+        var newChapterName = chapterName
+        try {
+            val chapterToInt = Integer.parseInt(chapterName)
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadProvider.kt:158]: There's a `String.toIntOrNull()` function that'd be useful in these kind of scenarios.

However, I don't think this is useful to do. If the chapter name is already just a number, then the downloaded chapter would already be fine?

Comment (Hipp0x) [reply app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadProvider.kt:158]: should we just close the issue and this PR if it's not useful?

Comment (arkon) [reply app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadProvider.kt:158]: The blank name is still applicable.

File: app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadProvider.kt (hunk 1 above)
Comment (Hipp0x) [reply app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadProvider.kt:160]: Do you think it's okay if I set a default value like "Chapter", or should I just find another way ?

Comment (arkon) [reply app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadProvider.kt:160]: That's fine.

Comment (Hipp0x) [reply app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadProvider.kt]: I've update the code

File: app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadProvider.kt [hunk 3]
```diff
@@ -138,14 +138,28 @@ class DownloadProvider(
...
         )
     }
 
+    /**
+     * Return the new name for the chapter (in case it's empty or blank)
+     *
+     * @param chapterName the name of the chapter
+     */
+    private fun getNewChapterName(chapterName: String): String {
+        return if (chapterName.isEmpty() || chapterName.isBlank()) {
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadProvider.kt:156]: `isBlank` is a superset of `isEmpty`

File: i18n/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -446,6 +446,7 @@
     <string name="save_chapter_as_cbz">Save as CBZ archive</string>
     <string name="split_tall_images">Split tall images</string>
     <string name="split_tall_images_summary">Improves reader performance</string>
+    <string name="chapter">Chapter</string>
```
Comment (arkon) [inline i18n/src/main/res/values/strings.xml:449]: Remove this.

File: app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadProvider.kt [hunk 5]
```diff
@@ -138,14 +138,28 @@ class DownloadProvider(
...
             },
         )
     }
 
+    /**
+     * Return the new name for the chapter (in case it's empty or blank)
+     *
+     * @param chapterName the name of the chapter
+     */
+    private fun getNewChapterName(chapterName: String): String {
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadProvider.kt:155]: `sanitizeChapterName` maybe

**Review by arkon — CHANGES_REQUESTED** [review]

**Comment by arkon** [comment]
I'll merge this on the basis that it's a _little_ bit better. It'll only realistically handle the case where there's _one_ chapter that's affected since you'll end up with conflicting names if there's multiple chapters that end up with a blank name normally.

Realistically, that should be fixed "properly" at the source/extension level.

Otherwise, the better solution would be #9238.

**Pull request merged**